packages/blade/src/components/BottomSheet/_decisions/decisions.md

```diff
@@ -0,0 +1,298 @@
+    leading={<BladeIcon />}
+  />
+  <BottomSheetBody>
+    <Text>Body Content<Text>
+  </BottomSheetBody>
+  <BottomSheetFooter
+    title="Footer Title"
+    leading={BladeIcon}
+    trailing={
+      <>
+        BladeIcon | IconButton | Button
+      </>
+    }
+  />
+</BottomSheet>;
+```
+
+**Design Constraints**
+
+- BottomSheetHeader:
+  - title
+  - leading:
+    - BladeIcon
+
+- BottomSheetFooter:
+  - title
+  - leading:
+    - BladeIcon
+  - trailing
+    - any jsx element
+
+- The Header, Footer can be individually omitted
+
+### `BottomSheet`
+
+We'll expose a `BottomSheet` component with the following API:
+
+| Prop            | Type        | Default                 | Description                                                                                     | Required |
+| --------------- | ----------- | ----------------------- | ----------------------------------------------------------------------------------------------- | -------- |
+| isOpen            | `boolean`   | `false`                 | Toggles bottom sheet state                                                              |          |
+| snapPoints      | `[number, number, number]`  | `[0.35, 0.5, 0.85]` | Snappoints in which the bottom sheeet will rest on, this accepts a number between 0 & 1 which maps to the total view height of the screen. 0.5 means 50% of screen height.                                              |          |
+| onDismiss       | `Callback`  | `undefined`             | called when the bottom sheet is closed, either by user state, hitting `esc` or tapping backdrop |          |
+| initialFocusRef | `React.Ref` | `undefined`             | ref element you want to get keyboard focus when opening the sheet                               |          |
+
+### `BottomSheetBody`
+
+| Prop       | Type              | Default     | Description                 | Required |
+| ---------- | ----------------- | ----------- | --------------------------- | -------- |
+| `children` | `React.ReactNode` | `undefined` | Contents of the BottomSheet | ✅       |
+
+### `BottomSheetHeader` API
+
+| Prop     | Type            | Default     | Description                          | Required |
+| -------- | --------------- | ----------- | ------------------------------------ | -------- |
+| `title`  | `string`        | `undefined` | Title of the Header                    | ✅       |
+| `leading` | `IconComponent` | `undefined` | leading icon placed before title text |          |
+
+### `BottomSheetFooter` API
+
+| Prop     | Type            | Default     | Description                          | Required |
+| -------- | --------------- | ----------- | ------------------------------------ | -------- |
+| `title`  | `string`        | `undefined` | Title of the BottomSheet footer      | ✅       |
+| `leading` | `IconComponent` | `undefined` | leading icon placed before title text |          |
+| `trailing` | `React.ReactNode` | `undefined` | Renders trailing content |          |
```

Comment:
How would we render such cases?

![image](https://user-images.githubusercontent.com/6682655/231782642-05d3f177-7a5f-4ed2-95d3-a8df44f0cce3.png)

I suspect people might use trailing for rendering everything 😆

Reply:
This is what is pending from design side, we need to think of a better unified API for all the header footers.

Reply:
Resolved.